lib/workers/schedule_worker.rb

```diff
@@ -74,150 +79,111 @@ def worker_setting_or_default(keys, default=nil)
-      @queue.enq :vmdb_database_log_all_database_statistics
+    builder.schedule_every(every, :tags => [:log_all_database_statistics]) do
+      enqueue(:vmdb_database_log_all_database_statistics)
     end
 
     # Schedule - Update Server Statistics
     every = worker_setting_or_default(:server_stats_interval)
-    @schedules[:all] << self.system_schedule_every(
-      every,
-      :first_in => every,
-      :tags     => [:status_update, schedule_category]
-    ) { @queue.enq :miq_server_status_update }
+    builder.schedule_every(every, :first_in => every, :tags => [:status_update]) { enqueue(:miq_server_status_update) }
 
     # Schedule - Log Server and Worker Statistics
     every = worker_setting_or_default(:server_log_stats_interval)
-    @schedules[:all] << self.system_schedule_every(
-      every,
-      :first_in => every,
-      :tags     => [:log_status, schedule_category]
-    ) { @queue.enq :miq_server_worker_log_status }
+    builder.schedule_every(every, :first_in => every, :tags => [:log_status]) { enqueue(:miq_server_worker_log_status) }
 
     # Schedule - Periodic logging of database statistics
     interval = worker_setting_or_default(:db_diagnostics_interval, 30.minutes)
-    @schedules[:all] << self.system_schedule_every(
-      interval,
-      :first_in => 1.minute,
-      :tags     => [:log_statistics, schedule_category]
-    ) { @queue.enq :miq_db_config_log_statistics }
+    builder.schedule_every(interval, :first_in => 1.minute, :tags => [:log_statistics]) do
+      enqueue(:miq_db_config_log_statistics)
+    end
 
     # Schedule - Periodic check for updates on appliances only
     if MiqEnvironment::Command.is_appliance?
       interval = worker_setting_or_default(:yum_update_check, 12.hours)
-      @schedules[:all] << self.system_schedule_every(
-        interval,
-        :first_in => 1.minute,
-        :tags     => [:server_updates, schedule_category]
-      ) { @queue.enq :miq_server_queue_update_registration_status }
+      builder.schedule_every(interval, :first_in => 1.minute, :tags => [:server_updates]) do
+        enqueue(:miq_server_queue_update_registration_status)
+      end
     end
 
     # Schedule - Periodic resync of RHN Mirror
     if MiqEnvironment::Command.is_appliance? && MiqServer.my_server.has_assigned_role?("rhn_mirror")
       interval = worker_setting_or_default(:resync_rhn_mirror, 12.hours)
-      @schedules[:all] << self.system_schedule_every(
-        interval,
-        :first_in => 1.minute,
-        :tags     => [:rhn_mirror, schedule_category]
-      ) { @queue.enq :miq_server_resync_rhn_mirror }
+      builder.schedule_every(interval, :first_in => 1.minute, :tags => [:rhn_mirror]) do
+        enqueue(:miq_server_resync_rhn_mirror)
+      end
     end
+
+    # These schedules need to be run on all servers regardless of the server's role
+    @schedules[:all] ||= []
```

Comment:
@imtayadeway `@schedules` is a tracking of the schedules that needs to synchronized with rufus.  If we add things to it, they need to be added to rufus.  If we want to remove them, we need to ensure we let [this code](https://github.com/imtayadeway/manageiq/blob/refactor-schedule-worker/lib/workers/schedule_worker.rb#L477-L495) do it so we ensure we remove it from rufus.  Note, I don't think that code is very great as it's not locking around the changes to the `@schedules` hash.  I don't recall if we have multiple threads at this point reading/editing this hash to know if it's important to lock around it.

Therefore we have to be careful about just dropping what's already in the hash for a specific key as we might drop the references to a schedule in the `@schedules` key but never remove them from rufus.  Note, you'd have to look at how each of these methods are called to see if you could ever get different schedules for each key.